### ISSUE TITLE
'Missing Push Notification Entitlement' for iOS standalone build

### ISSUE THREAD

**Issue by drpancake** [expo/expo#2092]
### Environment

```
  OS: macOS Sierra 10.12.6
  Node: 8.11.2
  Yarn: 1.7.0
  npm: 5.6.0
  Watchman: 4.7.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: Not Found
```

### Packages

```
  expo: ^28.0.0 => 28.0.1
  react: 16.3.1 => 16.3.1
  react-native: https://github.com/expo/react-native/archive/sdk-28.0.0.tar.gz => 0.55.4
```

### Steps to Reproduce

1. Create a standalone build using `exp build:ios` and allow Expo to manage all certificates and profiles
2. Upload the build to iTunes Connect

### Expected Behavior

The build is processed without issues and available for TestFlight submission.

### Actual Behavior

I get a warning email from iTunes Connect:

```
Missing Push Notification Entitlement - Your app appears to register with the Apple Push Notification service, but the app signature's entitlements do not include the "aps-environment" entitlement.

If your app uses the Apple Push Notification service, make sure your App ID is enabled for Push Notification in the Provisioning Portal, and resubmit after signing your app with a Distribution provisioning profile that includes the "aps-environment" entitlement.

Xcode 8 does not automatically copy the aps-environment entitlement from provisioning profiles at build time. This behavior is intentional. 
```

**Comment by webdevbyjoss** [comment]
Getting the same issue on the detached app (Create React Native App - "eject") that uses ExpoSDK.
Any ideas on how to solve this?

**Comment by webdevbyjoss** [comment]
In my case, the issue was caused by Firebase SDK in the app. It can be easily turned off, just follow this solution:
https://stackoverflow.com/questions/37428539/firebase-causes-issue-missing-push-notification-entitlement-after-delivery-to/43974297#43974297

**Comment by TPXP** [comment]
In order to fix this, you need to enable "Push notifications" for your app on the app store.
1. Login to https://developer.apple.com/
2. Go to the "Certificates, Identifiers and Profiles" section
3. Select "App IDs" in the Identifiers section on the left column
4. Choose your app ID, edit it and check "Push Notifications". Save changes
5. I'm not sure if it's needed, but just in case generate a certificate for the production push service: go to the "Certificates" section, and follow the assistant to add a "Apple Push Notification service SSL (Sandbox & Production)" certificate for your app. Acute readers will notice that expo does not need this certificate since it uses a key to talk to Apple directly

TL;DR: Expo needs to enable the "Push Notification" service when creating the app in the Apple Developer Portal

**Comment by tan116** [comment]
Hey TPXP.

I did what you said. I went and checked "Push Notifications" under the app id section and submitted another build. Got rejected within minutes for the same reason again.

**Comment by andregufc** [comment]
@TPXP  I got rejected too after the steps. The 5th step is needed.

I didn't get the last part about Expo. Where exactly can I enable that?
Thank you

**Comment by TPXP** [comment]
Thanks for your feedback @andregufc, I updated my comment.

For now, you cannot ask expo to enable Push Notifications when it creates the app on your Apple developer account, I just gave a hint for a fix. :wink:

**Comment by andregufc** [comment]
Finally I got success uploading the app. Thanks @TPXP !

Just for reference: I removed all the previous certificates and the provisioning profile related to my app. Then I followed the 5 steps again.
After that I did run: expo build:ios --clear-push-cert
After the building I uploaded my .ipa file through the Application Loader.

**Comment by vmheijke** [comment]
I have the same problem. I do not use Push notifications. In developer.apple.com it states that Push Notifications are disabled. 
Do I understand it correctly that I need to enable them to work with Expo (although I do not use it) or do I need to change something in my development environment? and if the second is true, what?
